From f920164998e475ee8dc6a3dc2be9d406c380acf3 Mon Sep 17 00:00:00 2001
From: Zhu Donghong <zhudonghong@wxiat.com>
Date: Wed, 18 May 2022 15:24:50 +0800
Subject: [PATCH] sw64: add builtin LPC interrupt controller to chip3.dts
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: f920164998e475ee8dc6a3dc2be9d406c380acf3
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I5G7AK

--------------------------------

This adds corresponding devicetree binding for LPC interrupt
controllers on chip3 series cpu.

Signed-off-by: Zhu Donghong <zhudonghong@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/boot/dts/chip3.dts | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/arch/sw_64/boot/dts/chip3.dts b/arch/sw_64/boot/dts/chip3.dts
index be2e91ee3279..082506393ac9 100644
--- a/arch/sw_64/boot/dts/chip3.dts
+++ b/arch/sw_64/boot/dts/chip3.dts
@@ -32,12 +32,20 @@ spiclk: spiclk {
 
 		};
 
-		intc: interrupt-controller{
+		intc: interrupt-controller {
 			compatible = "sw64,sw6_irq_controller";
 			interrupt-controller;
 			#interrupt-cells = <1>;
 		};
 
+		lpc_intc: interrupt-controller@0x8037 {
+			compatible = "sw64,lpc_intc";
+			reg = <0x8037 0x40000000 0x0 0x8000>;
+			interrupt-controller;
+			#interrupt-cells = <1>;
+			interrupt-parent = <&intc>;
+			interrupts = <2>;
+		};
 
 		uart: serial0@8033 {
 			#address-cells = <2>;
@@ -176,10 +184,8 @@ partition@0 {
 		lpc: lpc@0x8037 {
 			#address-cells = <2>;
 			#size-cells = <2>;
-			compatible = "sw,sw6b_lpc";
+			compatible = "sunway,chip3_lpc";
 			reg = <0x8037 0x40000000 0x0 0x8000>;
-			interrupt-parent=<&intc>;
-			interrupts = <2>;
 			status = "okay";
 
 		};
@@ -202,6 +208,8 @@ ipmi-bt@0x8037 {
 			device_type = "ipmi";
 			compatible = "ipmi-bt";
 			reg = <0x8037 0x100000e4 0x0 0x10>;
+			interrupt-parent=<&lpc_intc>;
+			interrupts = <10>;
 			reg-size = <1>;
 			reg-spacing = <1>;
 			reg-shift = <0>;
-- 
2.33.0

